### PR TITLE
Support for Tensorflow 2.17

### PR DESCRIPTION
**Pull request by mhuen**
Adds support for newest tensorflow version. This PR includes two updates:
 -  Convert KerasVariable to tensor
 - fix eager execution for LR scheduler

### CODE REVIEW

**Comment by mhuen** [comment]
Note: while falling back to compat.v1 optimizers works, support for variable learning rates is lost that was previously available. The issue is that newer tensorflow optimizers utilize keras optimizers which do not return the update operation when calling the `apply_gradients` method. 
A proper solution is to convert everything to TF2 instead of relying on `tf.compat.v1` symbols. See also #7 . This requires and extensive rewrite though.

**Comment by mhuen** [comment]
`tf.compat.v1` symbols are now removed and everything is updated to proper tensorflow 2 logic. This fixes issue #7 .
The documentation has now also been updated to reflect these changes. This PR will thus also resolve issues #3 and #4.
